CMakeLists.txt

```diff
@@ -573,128 +570,33 @@ if(QUDA_QDPJIT)
 endif()
 
 if(QUDA_QMP)
-  find_library(QMP_FOUND qmp HINTS ${QUDA_QMPHOME}/lib NO_DEFAULT_PATH)
-  mark_as_advanced(QMP_FOUND)
-  set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${QMP_FOUND})
-  if(QUDA_DOWNLOAD_USQCD AND NOT QMP_FOUND)
-    ExternalProject_Add(
-      QMP
-      GIT_REPOSITORY https://github.com/usqcd-software/qmp.git
-      GIT_TAG qmp2-5-2
-      GIT_SHALLOW YES
-      PREFIX usqcd
-      CONFIGURE_COMMAND CC=${MPI_C_COMPILER} CXX=${MPI_CXX_COMPILER} <SOURCE_DIR>/configure "INSTALL=${INSTALL_EXE} -C"
-                        "CFLAGS=-Wall -O3 -std=c99 " --with-qmp-comms-type=MPI --prefix=<INSTALL_DIR>
-      BUILD_COMMAND ${MAKE_EXE}
-      INSTALL_COMMAND ${MAKE_EXE} install
-      LOG_INSTALL ON
-      LOG_BUILD ON
-      LOG_DOWNLOAD ON)
-
-    ExternalProject_Get_Property(QMP INSTALL_DIR)
-    set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS ${INSTALL_DIR})
-    set(QUDA_QMPHOME
-        ${INSTALL_DIR}
-        CACHE PATH "path to QMP" FORCE)
-    set(QUDA_QMP_LDFLAGS
-        "-L${QUDA_QMPHOME}/lib"
-        CACHE STRING "LDFLAGS for QMP - should be derived from qmp-config --ldflags")
-    set(QUDA_QMP_LIBS
-        "-lqmp"
-        CACHE STRING "LIBS for QMP - should be derived from qmp-config --libs")
-    ExternalProject_Add_Step(
-      QMP reconf
-      COMMAND ${AUTORECONF_EXE} -fi
-      WORKING_DIRECTORY <SOURCE_DIR>
-      DEPENDERS configure
-      DEPENDEES download)
-
+  message(STATUS "Looking for QMP")
+  find_package(QMP QUIET) 
+  if( QMP_FOUND ) 
+    message(STATUS "Found QMP")
   else()
-    if("${QUDA_QMPHOME}" STREQUAL "")
-      message(FATAL_ERROR "QUDA_QMPHOME must be defined if QUDA_QMP is ON and QUDA_DOWNLOAD_USQCD is OFF")
-    endif()
-    execute_process(COMMAND ${QUDA_QMPHOME}/bin/qmp-config --cflags OUTPUT_VARIABLE QUDA_QMP_CFLAGS
-                                                                                    OUTPUT_STRIP_TRAILING_WHITESPACE)
-    execute_process(COMMAND ${QUDA_QMPHOME}/bin/qmp-config --ldflags OUTPUT_VARIABLE QUDA_QMP_LDFLAGS_INTERNAL
-                                                                                     OUTPUT_STRIP_TRAILING_WHITESPACE)
-    execute_process(COMMAND ${QUDA_QMPHOME}/bin/qmp-config --libs OUTPUT_VARIABLE QUDA_QMP_LIBS_INTERNAL
-                                                                                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-    set(QUDA_QMP_LDFLAGS
-        ${QUDA_QMP_LDFLAGS_INTERNAL}
-        CACHE STRING "LDFLAGS for QMP - should be derived from qmp-config --ldflags")
-    set(QUDA_QMP_LIBS
-        ${QUDA_QMP_LIBS_INTERNAL}
-        CACHE STRING "LIBS for QMP - should be derived from qmp-config --libs")
+    message(STATUS "QMP not found")
+  endif()
+
+  if(QUDA_DOWNLOAD_USQCD AND NOT QMP_FOUND)
```

Comment:
find_package() won't find QMP until it is built. So we can't use it to find a dependency that is downloaded using `External_Project_Add`. We could use `FetchContent` to a subdir and add that subdir tho. Or we can use a package manager. So for now I disabled this